### PR TITLE
Hide option to show braille messages indefinitely when messages timeout is set to 0

### PR DESCRIPTION
**Pull request by dawidpieper**
### Summary of the issue:
When user sets braille messages timeout to 0, option to show them indefinitely still appears. Its checking does not take any effect, though. I found it confusing for some people.

### Description of how this pull request fixes the issue:
Now there is a combobox that enables users to set if braille messages should be displayed and for how long:
* Disabled - disables showing of braille messages
* Use timeout - Enables users to set custom timeout
* Show indefinitely - Messages never disappear automatically.

Checkbox to show messages indefinitely has been removed and timeout field is now shown only if "Use timeout" option is selected.

### Testing performed:
All tests passed.

### Known issues with pull request:
None yet

### Change log entry:

Changes
Rebuilt option to set if braille messages should be shown and when they should disappear

### CODE REVIEW

**Comment by feerrenrut** [comment]
It may not be immediately obvious why this control is missing.

Instead, perhaps the following approach:
- Use a comboBox "Show braille messages", with options: "Disable", "Use timeout", "Show indefinitely"
- When "Show indefinitely" is selected, then the timeout control is enabled, otherwise it is disabled.

Also, please check the user guide and confirm that it describes these options well (or needs changes).

**Comment by dawidpieper** [comment]
Thank you, I think it would be a good approach. I'll try to prepare it tomorrow.

**Comment by XLTechie** [comment]
I'm slightly confused by this:

>  *  Use a comboBox "Show braille messages", with options: "Disable", "Use timeout", "Show indefinitely"
>  *  When "Show indefinitely" is selected, then the timeout control is enabled, otherwise it is disabled.

Why would the timeout control be available while "Show indefinitely" is 
selected, instead of while "Use timeout" is selected?

If I was using this option, I would expect "Use timeout" to give me control over 
what the timeout was, not the other way around.

Now, I don't use any braille options, so I am maybe missing an important factor, 
but just thinking about it logically.

**Comment by dawidpieper** [comment]
I may be wrong, but I suppose it was a small error in the comment.

**Comment by feerrenrut** [comment]
> If I was using this option, I would expect "Use timeout" to give me control over
what the timeout was, not the other way around.

Yes, you are correct. I guess I was thinking one thing and typing another. I'll edit the comment to avoid any future confusion.

File: source/gui/settingsDialogs.py [hunk 1]
```diff
@@ -3257,13 +3257,42 @@ def makeSettings(self, settingsSizer):
...
+		elif config.conf["braille"]["noMessageTimeout"] == 0:
+			self.showMessagesList.SetSelection(1)
+		else:
+			self.showMessagesList.SetSelection(2)
+
 		minTimeout = int(config.conf.getConfigValidation(
 			("braille", "messageTimeout")
 		).kwargs["min"])
+		if minTimeout == 0:
+			minTimeout = 1
```
[the remark this answers is not in the document]
Comment (feerrenrut) [reply source/gui/settingsDialogs.py:3292]: If you decide to keep it like this, it is probably a good idea to add a comment to the config spec to document what 0 means. Search for `messageTimeout` in the config spec to find it.

**Review by feerrenrut — CHANGES_REQUESTED** [review]
Generally looks good, thanks @dawidpieper

**Comment by dawidpieper** [comment]
I've harcoded 1 as a minimum value. I don't know what is the main policy for config specs - readability or simplicity. I can also change config specs and prepare upgrade as suggested, if it is the preferred way.

**Review by feerrenrut — APPROVED** [review]
Thanks @dawidpieper

**Pull request merged**